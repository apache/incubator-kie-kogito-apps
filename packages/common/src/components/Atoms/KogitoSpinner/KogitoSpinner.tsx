import React from 'react';
import {
  Title,
  EmptyState,
  EmptyStateIcon,
  Spinner
} from '@patternfly/react-core';

interface KogitoSpinnerProps {
  spinnerText: string;
}
const KogitoSpinner: React.FC<KogitoSpinnerProps> = ({ spinnerText }) => {
  return (
    <EmptyState>
      <EmptyStateIcon variant="container" component={Spinner} />
      <Title size="lg">{spinnerText}</Title>
    </EmptyState>
  );
};

KogitoSpinner.displayName = 'KogitoSpinner';
export default KogitoSpinner;
