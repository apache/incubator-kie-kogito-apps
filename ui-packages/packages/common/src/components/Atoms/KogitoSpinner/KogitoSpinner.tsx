import React from 'react';
import {
  Title,
  EmptyState,
  EmptyStateIcon,
  Spinner
} from '@patternfly/react-core';
import { OUIAProps, componentOuiaProps } from '../../../utils/OuiaUtils';

interface KogitoSpinnerProps {
  spinnerText: string;
}
const KogitoSpinner: React.FC<KogitoSpinnerProps & OUIAProps> = ({
  spinnerText,
  ouiaId,
  ouiaSafe
}) => {
  return (
    <EmptyState {...componentOuiaProps(ouiaId, 'kogito-spinner', ouiaSafe)}>
      <EmptyStateIcon variant="container" component={Spinner} />
      <Title size="lg">{spinnerText}</Title>
    </EmptyState>
  );
};

KogitoSpinner.displayName = 'KogitoSpinner';
export default KogitoSpinner;
