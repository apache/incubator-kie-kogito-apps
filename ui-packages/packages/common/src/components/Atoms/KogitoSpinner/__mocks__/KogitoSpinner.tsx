import React from 'react';

const MockedKogitoSpinner = () => {
  return <></>;
};

export default MockedKogitoSpinner;
