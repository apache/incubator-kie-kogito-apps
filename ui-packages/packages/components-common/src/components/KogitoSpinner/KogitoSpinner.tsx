import React from 'react';
import {
  EmptyState,
  EmptyStateIcon
} from '@patternfly/react-core/dist/js/components/EmptyState';
import { Title } from '@patternfly/react-core/dist/js/components/Title';
import { Spinner } from '@patternfly/react-core/dist/js/components/Spinner';
import {
  OUIAProps,
  componentOuiaProps
} from '@kogito-apps/ouia-tools/dist/utils/OuiaUtils';

interface KogitoSpinnerProps {
  spinnerText: string;
}
export const KogitoSpinner: React.FC<KogitoSpinnerProps & OUIAProps> = ({
  spinnerText,
  ouiaId,
  ouiaSafe
}) => {
  return (
    <EmptyState {...componentOuiaProps(ouiaId, 'kogito-spinner', ouiaSafe)}>
      <EmptyStateIcon variant="container" component={Spinner} />
      <Title size="lg" headingLevel="h3">
        {spinnerText}
      </Title>
    </EmptyState>
  );
};
