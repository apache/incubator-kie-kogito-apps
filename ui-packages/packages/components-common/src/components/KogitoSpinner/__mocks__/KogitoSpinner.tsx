import React from 'react';

const MockedKogitoSpinner: React.FC = () => {
  return <></>;
};

export default MockedKogitoSpinner;
