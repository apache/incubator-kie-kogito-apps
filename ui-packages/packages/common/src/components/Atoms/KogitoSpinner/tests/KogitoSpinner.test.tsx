import React from 'react';
import { shallow } from 'enzyme';
import KogitoSpinner from '../KogitoSpinner';

describe('KogitoSpinner component tests', () => {
  it('snapshot testing with loading test', () => {
    const wrapper = shallow(
      <KogitoSpinner
        spinnerText={'loading...'}
        ouiaId="kogito-spinner-ouia-id"
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
