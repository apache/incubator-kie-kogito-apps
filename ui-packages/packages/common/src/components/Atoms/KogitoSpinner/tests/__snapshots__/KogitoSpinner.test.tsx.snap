// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`KogitoSpinner component tests snapshot testing with loading test 1`] = `
<EmptyState
  data-ouia-component-id="kogito-spinner-ouia-id"
  data-ouia-component-type="kogito-spinner"
  data-ouia-safe={true}
>
  <EmptyStateIcon
    component={[Function]}
    variant="container"
  />
  <Title
    size="lg"
  >
    loading...
  </Title>
</EmptyState>
`;
